mono/metadata/object-internals.h

```diff
@@ -572,7 +572,7 @@ struct _MonoThreadInfo;
 
 typedef struct MonoThreadName {
 	char* volatile chars;      // null check outside of lock
-	gsize volatile generation; // read outside of lock
+	gsize volatile generation; // hypothetically read outside of lock; no longer actually used
```

Comment:
Can we remove it?

Reply:
Yes we could. It is a reasonable idea, but yes, removable.
It predates the ability to just do the pointer compare.

Reply:
To be clear, w/o this PR, it is used. This PR replaces the use.

Reply:
IMHO it should be removed if it's unused. We can always bring it back from history if necessary. Otherwise the change LGTM.

Reply:
I'd be resolving merge conflicts in configure.ac every few days for months, had I made that change.

Reply:
I’ll remove it today. Hopefully without having it sit months longer.